dspy/utils/usage_tracker.py

```diff
@@ -20,16 +20,16 @@ def __init__(self):
         # }
         self.usage_data = defaultdict(list)
 
-    def _flatten_usage_entry(self, usage_entry) -> dict[str, dict[str, Any]]:
+    def _flatten_usage_entry(self, usage_entry: dict[str, Any]) -> dict[str, Any]:
```

Comment:
`usage_entry` is not necessarily a dict as i remember, it could be the litellm usage object so I skipped type hints for this method (and the others)